src/main/java/one/util/streamex/DoubleStreamEx.java

```diff
@@ -580,6 +582,61 @@ public DoubleStreamEx skipOrdered(long n) {
         return delegate(spliterator);
     }
 
+    /**
+     * Returns a new, non-splittable, sequential stream containing
+     * all the elements of the original stream except the last n elements.
+     * Consumes space proportional to the number of elements skipped.
+     *
+     * <p>
+     * For example, {@code DoubleStreamEx.of(1d, 2d, 3d).skipLast(1)} will yield a stream containing
+     * two elements: 1d, 2d.
+     *
+     * <p>
+     * This is an <a href="package-summary.html#StreamOps">quasi-intermediate operation</a>.
+     *
+     * @param n the non-negative number of elements to leave off the end of the stream.
+     * @return the new, sequential stream
+     * @throws IllegalArgumentException if {@code n} is negative or {@link Integer#MAX_VALUE}
+     * @since 0.6.7
+     */
+    public DoubleStreamEx skipLast(int n) {
+        if (n < 0 || n == Integer.MAX_VALUE)
+            throw new IllegalArgumentException(Long.toString(n));
+        if (n == 0)
+            return this;
+
+        BlockingDeque<Double> buffer = new LinkedBlockingDeque<>(n + 1);

```

Comment:
Unnecessary boxing is absolutely not an option. You could just create an array of n elements.